lib/puppet/ssl/validator/default_validator.rb

```diff
@@ -100,7 +100,6 @@ def call(preverify_ok, store_context)
           @verify_errors << error_string
         end
       else
-        current_cert = store_context.current_cert
```

Comment:
I discovered a strange thing happening here in our unit tests, where apparently at least with the way we are mocking this (https://github.com/Magisus/puppet/blob/master/spec/unit/ssl/validator_spec.rb#L35) if you query `store_context.current_cert` twice, it returns a different thing the second time (maybe the next one in the list?). I wasn't sure why this was happening, but removing the duplicate call fixed it. It seemed wrong to set `current_cert` twice anyway.